### PR TITLE
[4.0] Fixing ordering of maps in smart search

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #30965.

### Summary of Changes
The maps view is not properly ordering the content, which is mainly because the table has been changed from a flat list to a nested table. Since the rows aren't ordered correctly, tasks like publishing the content are failing. This in turn is due to the code checking if parents are published and if the list of items to publish is not in the right order (parents first), that check fails and the whole process fails. This fixes the ordering (more or less) and thus hopefully fixes the publishing issue for now.

### Testing Instructions
1. Install Joomla, install some sampledata.
2. Go to Smart Search -> Maps
3. See that the rows are somehow strange, messed up, chaotic.
4. Select all maps and unpublish and then publish again. The later should fail.
5. Apply patch
6. See that for example authors show all up below "Authors"
7. Select all maps and click on publishing. Now this works.

### CODE REVIEW

File: administrator/components/com_finder/src/Model/MapsModel.php [hunk 1]
```diff
@@ -228,7 +228,7 @@ protected function getListQuery()
 		}
 
 		// Add the list ordering clause.
-		$query->order($db->escape($this->getState('list.ordering', 'd.branch_title')) . ' ' . $db->escape($this->getState('list.direction', 'ASC')));
+		$query->order($db->escape($this->getState('list.ordering', 'branchtitle, a.lft')) . ' ' . $db->escape($this->getState('list.direction', 'ASC')));
```
Comment (brianteeman) [inline administrator/components/com_finder/src/Model/MapsModel.php]: shouldn't this be branch_title ?

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on a22b8e7c0cf5a88d586188975196b4c67068cd74<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33885">issues.joomla.org/tracker/joomla-cms/33885</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 2436fd707661f65d4bc8f62ad7c5bba19cd5a541<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33885">issues.joomla.org/tracker/joomla-cms/33885</a>.</sub>

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 2436fd707661f65d4bc8f62ad7c5bba19cd5a541<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33885">issues.joomla.org/tracker/joomla-cms/33885</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33885">issues.joomla.org/tracker/joomla-cms/33885</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**